src/test/java/org/opensearch/neuralsearch/search/query/HybridQueryPhaseSearcherTests.java

```diff
@@ -0,0 +1,387 @@
+import org.opensearch.index.Index;
+import org.opensearch.index.mapper.TextFieldMapper;
+import org.opensearch.index.query.MatchAllQueryBuilder;
+import org.opensearch.index.query.QueryBuilder;
+import org.opensearch.index.query.QueryBuilders;
+import org.opensearch.index.query.QueryShardContext;
+import org.opensearch.index.query.TermQueryBuilder;
+import org.opensearch.index.shard.ShardId;
+import org.opensearch.knn.index.mapper.KNNVectorFieldMapper;
+import org.opensearch.neuralsearch.query.HybridQueryBuilder;
+import org.opensearch.neuralsearch.query.OpenSearchQueryTestCase;
+import org.opensearch.neuralsearch.search.CompoundTopDocs;
+import org.opensearch.search.SearchShardTarget;
+import org.opensearch.search.internal.ContextIndexSearcher;
+import org.opensearch.search.internal.SearchContext;
+import org.opensearch.search.query.QueryCollectorContext;
+import org.opensearch.search.query.QuerySearchResult;
+
+import com.carrotsearch.randomizedtesting.RandomizedTest;
+
+public class HybridQueryPhaseSearcherTests extends OpenSearchQueryTestCase {
+    private static final String VECTOR_FIELD_NAME = "vectorField";
+    private static final String TEXT_FIELD_NAME = "field";
+    private static final String TEST_DOC_TEXT1 = "Hello world";
+    private static final String TEST_DOC_TEXT2 = "Hi to this place";
+    private static final String TEST_DOC_TEXT3 = "We would like to welcome everyone";
+    private static final String TEST_DOC_TEXT4 = "This is really nice place to be";
+    private static final String QUERY_TEXT1 = "hello";
+    private static final String QUERY_TEXT2 = "randomkeyword";
+    private static final String QUERY_TEXT3 = "place";
+    private static final Index dummyIndex = new Index("dummy", "dummy");
+    private static final String MODEL_ID = "mfgfgdsfgfdgsde";
+    private static final int K = 10;
+    private static final QueryBuilder TEST_FILTER = new MatchAllQueryBuilder();
+
+    @SneakyThrows
+    public void testQueryType_whenQueryIsHybrid_thenCallHybridDocCollector() {
+        HybridQueryPhaseSearcher hybridQueryPhaseSearcher = spy(new HybridQueryPhaseSearcher());
+        QueryShardContext mockQueryShardContext = mock(QueryShardContext.class);
+        KNNVectorFieldMapper.KNNVectorFieldType mockKNNVectorField = mock(KNNVectorFieldMapper.KNNVectorFieldType.class);
+        when(mockQueryShardContext.index()).thenReturn(dummyIndex);
+        when(mockKNNVectorField.getDimension()).thenReturn(4);
+        when(mockQueryShardContext.fieldMapper(eq(VECTOR_FIELD_NAME))).thenReturn(mockKNNVectorField);
+        TextFieldMapper.TextFieldType fieldType = (TextFieldMapper.TextFieldType) createMapperService().fieldType(TEXT_FIELD_NAME);
+        when(mockQueryShardContext.fieldMapper(eq(TEXT_FIELD_NAME))).thenReturn(fieldType);
+
+        Directory directory = newDirectory();
+        final IndexWriter w = new IndexWriter(directory, newIndexWriterConfig(new MockAnalyzer(random())));
```

Comment:
Some are final and others are not final. There is no consistency. Could be better to not to use final at all?

Reply:
ack